libp2p/crypto/crypto.nim

```diff
@@ -8,29 +8,79 @@
 ## those terms.
 
 ## This module implements Public Key and Private Key interface for libp2p.
-
 {.push raises: [Defect].}
+from strutils import split, strip, cmpIgnoreCase
+
+const libp2p_pki_schemes* {.strdefine.} = "rsa,ed25519,secp256k1,ecnist"
+
+type
+  PKScheme* = enum
+    RSA = 0,
+    Ed25519,
+    Secp256k1,
+    ECDSA
+
+proc initSupportedSchemes(list: static string): set[PKScheme] =
+  var res: set[PKScheme]
+
+  let schemes = split(list, {',', ';', '|'})
+  for item in schemes:
+    if cmpIgnoreCase(strip(item), "rsa") == 0:
+      res.incl(PKScheme.RSA)
+    elif cmpIgnoreCase(strip(item), "ed25519") == 0:
+      res.incl(PKScheme.Ed25519)
+    elif cmpIgnoreCase(strip(item), "secp256k1") == 0:
+      res.incl(PKScheme.Secp256k1)
+    elif cmpIgnoreCase(strip(item), "ecnist") == 0:
```

Comment:
I find ecnist vs ecdsa confusing

Reply:
@arnetheduck ECDSA its Elliptic Curve Digital Signature Algorithm and this algorithm can work with almost any curve, so for example signature and verification with bitcoin's secp256`k`1 is also ECDSA. But initially this algorithm was used with NIST curves which are secp256`r`1, secp384`r`1 and secp521`r`1. The name i have used initially `ECDSA` was incorrect but it was name which was used by `go` code.